cmake/bintools/gnu/target.cmake

```diff
@@ -2,16 +2,16 @@
 
 # Configures binary tools as GNU binutils
 
-find_program(CMAKE_OBJCOPY ${CROSS_COMPILE}objcopy PATHS ${TOOLCHAIN_HOME} NO_DEFAULT_PATH)
-find_program(CMAKE_OBJDUMP ${CROSS_COMPILE}objdump PATHS ${TOOLCHAIN_HOME} NO_DEFAULT_PATH)
-find_program(CMAKE_AS      ${CROSS_COMPILE}as      PATHS ${TOOLCHAIN_HOME} NO_DEFAULT_PATH)
-find_program(CMAKE_AR      ${CROSS_COMPILE}ar      PATHS ${TOOLCHAIN_HOME} NO_DEFAULT_PATH)
-find_program(CMAKE_RANLIB  ${CROSS_COMPILE}ranlib  PATHS ${TOOLCHAIN_HOME} NO_DEFAULT_PATH)
-find_program(CMAKE_READELF ${CROSS_COMPILE}readelf PATHS ${TOOLCHAIN_HOME} NO_DEFAULT_PATH)
-find_program(CMAKE_NM      ${CROSS_COMPILE}nm      PATHS ${TOOLCHAIN_HOME} NO_DEFAULT_PATH)
-find_program(CMAKE_STRIP   ${CROSS_COMPILE}strip   PATHS ${TOOLCHAIN_HOME} NO_DEFAULT_PATH)
+find_program(CMAKE_OBJCOPY ${CROSS_COMPILE}objcopy    PATHS ${TOOLCHAIN_HOME} NO_DEFAULT_PATH)
+find_program(CMAKE_OBJDUMP ${CROSS_COMPILE}objdump    PATHS ${TOOLCHAIN_HOME} NO_DEFAULT_PATH)
+find_program(CMAKE_AS      ${CROSS_COMPILE}as         PATHS ${TOOLCHAIN_HOME} NO_DEFAULT_PATH)
+find_program(CMAKE_AR      ${CROSS_COMPILE}gcc-ar     PATHS ${TOOLCHAIN_HOME} NO_DEFAULT_PATH)
+find_program(CMAKE_RANLIB  ${CROSS_COMPILE}gcc-ranlib PATHS ${TOOLCHAIN_HOME} NO_DEFAULT_PATH)
+find_program(CMAKE_READELF ${CROSS_COMPILE}readelf    PATHS ${TOOLCHAIN_HOME} NO_DEFAULT_PATH)
+find_program(CMAKE_NM      ${CROSS_COMPILE}nm         PATHS ${TOOLCHAIN_HOME} NO_DEFAULT_PATH)
+find_program(CMAKE_STRIP   ${CROSS_COMPILE}strip      PATHS ${TOOLCHAIN_HOME} NO_DEFAULT_PATH)
```

Comment:
FWIW: the use of static libraries internal to our build is almost all cargo-cult.  There are a tiny handful of places where we're using it as a trick to prevent the linker from seeing static data (like init calls and iterable elements) it would otherwise try to include.  All those should be correctly isolating the data via kconfig in one of the dozens of ways we allow and not doing it implicitly.